src/test/java/unit/GcsBucketFileReferenced.java

```diff
@@ -0,0 +1,432 @@
+package unit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.cli.serialization.userfacing.resource.UFGcsBucketFile;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.cloud.Identity;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.common.base.Optional;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.Auth;
+import harness.utils.ExternalGCSBuckets;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("unit")
+public class GcsBucketFileReferenced extends SingleWorkspaceUnit {
+
+  // external bucket to use for creating GCS bucket references in the workspace
+  private BucketInfo externalBucket;
+
+  // name of blob in external bucket
+  private String externalBucketBlobName = "blobs/testBlob";
+
+  @BeforeAll
+  @Override
+  protected void setupOnce() throws Exception {
+    super.setupOnce();
+    externalBucket = ExternalGCSBuckets.createBucket();
```

Comment:
I think we should test both cases:
- user has access to the bucket, and by extension a file in it
- user does not have full bucket access, only to a specific file in it

No need to duplicate all the tests for both cases, perhaps just for `add-ref` (which does an access check) and `describe` (which will try to lookup information directly from the cloud). If it's easier to put these in a separate class, similar to `GcsBucketNumObjects` unit test class.